clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -680,6 +790,7 @@ private long maybeUpdate(long now, Node node) {
         final RequestCompletionHandler callback;
         final boolean expectResponse;
         final boolean isInternalMetadataRequest; // used to flag metadata fetches which are triggered internally by NetworkClient
+        final boolean isInternalApiVersionRequest; // used to flag api version fetches which are triggered internally by NetworkClient
```

Comment:
Yeah, that's exactly what I was thinking. It seems like a nice improvement to let the `MetadataUpdater` interface accept an instance of `MetadataResponse` directly so that it doesn't have to deal with casting itself.